cram/cram_encode.c

```diff
@@ -1966,7 +1972,7 @@ int cram_encode_container(cram_fd *fd, cram_container *c) {
         kstring_t MD = {0};
 
         // Embed consensus / MD-generated ref
-        if (embed_ref == 2) {
+        if (embed_ref == 2 && c->ref_id >= 0) {
```

Comment:
I don't think `c->ref_id >= 0` can now happen due to the other changes, but if it did, could it cause problems elsewhere?  It looks like you might end up calling `process_one_read()` with `embed_ref = 2` but `c->ref = NULL`, which I'm not sure it can handle...  It might be safer to treat `c->ref_id < 0` like a `cram_generate_reference()` failure, so you switch firmly into non-ref mode.

Reply:
That's a simple enough change and safe to do I think.